integration/map-std/src/unstable.ts

```diff
@@ -167,13 +83,334 @@ export function resolveRequestUrl(url: string, options: { parameters: any, secur
+    } else {
+      this.headersList.set(lowercasedName, [value]);
+    }
+  }
+
+  // https://fetch.spec.whatwg.org/#dom-headers-delete
+  public delete(name: string): void {
+    this.headersList.delete(name.toLowerCase());
+  }
+
+  // https://fetch.spec.whatwg.org/#dom-headers-get
+  public get(name: string): string | null {
+    const values = this.headersList.get(name.toLowerCase());
+    if (values === undefined) {
+      return null;
+    }
+
+    return values.join(', ');
+  }
+
+  // https://fetch.spec.whatwg.org/#dom-headers-getsetcookie
+  public getSetCookie(): string[] {
+    return this.headersList.get('set-cookie') ?? [];
+  }
+
+  // https://fetch.spec.whatwg.org/#dom-headers-has
+  public has(name: string): boolean {
+    return this.headersList.has(name.toLowerCase());
+  }
+
+  // https://fetch.spec.whatwg.org/#dom-headers-set
+  public set(name: string, value: string): void {
+    this.headersList.set(name.toLowerCase(), [value]);
+  }
+
+  private fill(object: HeadersInit) {
+    if (Array.isArray(object)) {
+      for (const header of object) {
+        if (header.length !== 2) {
+          throw new TypeError(`Expected name/value pair to be length 2, found ${header.length}.`)
+        }
+
+        this.append(header[0], header[1]);
+      }
+    } else if (typeof object === 'object' && object !== null) {
+      for (const [name, value] of Object.entries(object)) {
+        this.append(name, value);
+      }
+    } else {
+      throw new TypeError("Headers must be: { string: string[] } | [string, string][]");
+    }
+  }
+}
+
+export type USVString = string;
+export interface RequestInit {
+  method?: string,
+  headers?: HeadersInit,
+  query?: MultiMap,
+  body?: BodyInit,
+  security?: string,
+}
+export type Request = RequestInit & { url: USVString };
+export type RequestInfo = USVString | Request;
+
+// https://fetch.spec.whatwg.org/#response-class
+type ResponseMessage = {
+  kind: 'ok';
+  status: number,
+  headers: Record<string, string[]>,
+  body_stream?: number;
+} | {
+  kind: 'err',
+  error_code: number,
+  message: string,
+};
+
+export type XMLHttpRequestBodyInit = Buffer | USVString; // TODO: Blob, BufferSource, FormData, URLSearchParams
+export type BodyInit = XMLHttpRequestBodyInit; // TODO ReadableStream
+
+export type ResponseInit = {
+  status?: number,
+  statusText?: string,
+  headers?: HeadersInit,
+};
+export enum ResponseType { 'basic', 'cors', 'default', 'error', 'opaque', 'opaqueredirect' };
+export class Response implements Body {
+  #handle: number;
+  #init: ResponseInit & { url: USVString };
+  #body: BodyInit | undefined;
+  #response: ResponseMessage | undefined;
+  #status: number | undefined;
+  #headers: Headers | undefined;
+  #bodyStream: ByteStream | undefined;
+
+  public readonly type: ResponseType;
+
+  constructor(
+    body?: BodyInit,
+    init?: ResponseInit,
+  ) {
+    this.#handle = -1;
+    this.#init = { ...init, url: '', status: 200, statusText: '', headers: {} };
```

Comment:
issue: This always overwrites the fields with defaults

Reply:
🤦